internal/backends/postgresql/collection.go

```diff
@@ -143,8 +143,59 @@ func (c *collection) InsertAll(ctx context.Context, params *backends.InsertAllPa
 
 // UpdateAll implements backends.Collection interface.
 func (c *collection) UpdateAll(ctx context.Context, params *backends.UpdateAllParams) (*backends.UpdateAllResult, error) {
-	// TODO https://github.com/FerretDB/FerretDB/issues/3391
-	return new(backends.UpdateAllResult), nil
+	p, err := c.r.DatabaseGetExisting(ctx, c.dbName)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if p == nil {
+		return nil, lazyerrors.Errorf("no database %q", c.dbName)
```

Comment:
This error is unreachable because collection is created earlier in https://github.com/FerretDB/FerretDB/blob/main/internal/handlers/sqlite/msg_update.go#L103

Added integration test for it too.

Reply:
Backend implementations should not follow what a command handler with a similar name does, because that backend/collection method could be called from multiple places; for example, `UpdateAll` is also called from `findAndModify` command handler.

Instead, both the handler and backend implementations should follow the contract: https://github.com/FerretDB/FerretDB/blob/8788b9212f3432ed7ac9204476fcc815d0492f16/internal/backends/collection.go#L146-L147

Please check other places for that, too.

Reply:
That is true. I updated implementation for sqlite as well :pray: